modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/PostGISDialect.java

```diff
@@ -1083,7 +1095,11 @@ public void encodeValue(Object value, Class type, StringBuffer sql) {
             else {
                 encodeByteArrayAsEscape(input, sql);
             }
-
+        } else if (type == PostGISDialect.XDate.class) {
```

Comment:
So I can kind of see how this works, XDate class is "just" a placeholder to mark "bigdate" and is being used to add in this special case.